bin/write_viral_gff.py

```diff
@@ -21,136 +20,213 @@ def get_ena_contig_mapping(ena_contig_file):
+    contig_protID={}
+    prot_data={}
+    contig_data={}
+    for vfile in virify_files:
+        with open(vfile,'r') as input_table:
+            next(input_table)
+            for line in input_table:
+                line_l=line.rstrip().split('\t')
+                Contig=line_l[0]
+                CDS_ID=line_l[1]
+                Start=line_l[2]
+                if Start == '0':
+                    Start=1
+                End=line_l[3]
+                Direction=line_l[4]
+                Best_hit=line_l[5]
+
+                if '|phage-circular' in Contig:
+                    Contig=Contig.replace('|phage-circular','')
+                    CDS_ID=CDS_ID.replace('|phage-circular','')
+                    desc='phage_circular'
+
+                elif '|prophage-' in Contig:
+                    desc='prophage'
+                    # Fixing CDS coordinates to the context of the whole contig
+                    prophage_start=int(Contig.split('|')[1].split('-')[1].split(':')[0])
+                    Start=int(Start)+prophage_start
+                    End=int(End)+prophage_start
+
                 else:
-                    if contigs[row["Contig"]]["start"] > row["Start"]:
-                        contigs[row["Contig"]]["start"] = row["Start"]
-                    if contigs[row["Contig"]]["end"] < row["End"]:
-                        contigs[row["Contig"]]["end"] = row["End"]
-
-                #   remove '.faa' trailing viphog ID
-                viphog = row["Best_hit"].strip(".faa")
-                direction = "-" if row["Direction"] == "-1" else "+"
-
-                #   change contig name if ena mapping required
-                if ena_mapping:
-                    contig_name = f'{ena_mapping[row["Contig"]]}-{row["Contig"]}'
+                    Contig=Contig.replace('|','')
+                    CDS_ID=CDS_ID.replace('|','')
+                    desc='phage_linear'
+
+                if Contig not in contig_protID.keys():
+                    contig_protID[Contig]=[CDS_ID]
+                    contig_data[Contig]=desc
                 else:
-                    contig_name = row["Contig"]
-
-                #   ID=ERZ2271866.1-NODE-1-length-21396-cov-5.122534;viphog=ViPhOG1;viphog_taxonomy=Phaeovirus
-                annotation = (
-                    f'ID={contig_name};viphog={viphog};viphog_taxonomy={row["Label"]}'
-                )
-                #   start with: ERZ2271866.1-NODE-1-length-21396-cov-5.122534	ViPhOG	proviral_region	1020	2050	.	-	.
-                gff.write(
-                    f'{contig_name}\t{viphog}\tviral_sequence\t{row["Start"]}\t{row["End"]}\t.\t{direction}'
-                    f"\t.\t{annotation}\n"
-                )
-    return contigs
-
-
-def write_metadata(
-    checkv_files, taxonomy_files, sample_prefix, virify_contigs, ena_mapping=None
+                    contig_protID[Contig].append(CDS_ID)
+
+                Direction=Direction.replace('-1','-').replace('1','+')
```

Comment:
We use `snake_case` as the naming convention. Variables should not be un uppercase, `Direction` should be `direction`

Reply:
I've changed all the variables starting with uppercase for lowercase